federation/apis/federation/register.go

```diff
@@ -48,7 +48,6 @@ func addKnownTypes(scheme *runtime.Scheme) error {
 		&Cluster{},
 		&ClusterList{},
 		&api.ListOptions{},
-		&api.DeleteOptions{},
```

Comment:
I experienced a similar error when I have `DeleteOptions` there. The error went away when I remove it. I found there is no such option in other register.go files.
This registing mechanism seems behaving weird as I found sometimes it get overridden or conflicted for the same type in different group version. I haven't got time to look into the root cause however.
@nikhiljindal